### PR TITLE
Feat(code): Cli arg processing with pico-args

### LINKED ISSUE #20: consider using pico-args and --version flag
https://github.com/RazrFalcon/pico-args says it adds 0.5s to crate compilation time, which feels like it is probably within budget.

Adding support for --version feels like a good idea, but I don't want to roll my own arg-parser.

### PR DESCRIPTION
**Pull request by progga**
Manages cli arguments using the pico-args crate.  Also added --version flag
support for disclosing own version.

Closes #20

### CODE REVIEW

**Comment by progga** [comment]
Hello,
I have landed here from today's [London Rust Hack and Learn](https://github.com/rust-ldn/rust-hack-and-learn/issues/15) :)

**Comment by alsuren** [comment]
Oh Hi. I finally made it to the event. Do you want to jump in and pair on reviewing it?

**Comment by alsuren** [comment]
https://app.wonder.me/?spaceId=84079fa7-d7cb-4768-95fd-3802d5fb4ed0

File: cargo-quickinstall/src/main.rs [hunk 1]
```diff
@@ -285,14 +286,21 @@ fn install_crate(crate_name: &str, version: &str, target: &str) -> Result<(), In
...
-    } else {
-        args.get(1)
-    };
-
-    let crate_name = crate_name.ok_or("USAGE: cargo quickinstall CRATE_NAME")?;
+
+    let mut args = Arguments::from_env();
+    if args.contains("--version") {
+        let version_self = env!("CARGO_PKG_VERSION");
+        println!("{}", version_self);
```
Comment (alsuren) [inline cargo-quickinstall/src/main.rs:293]: Ah. I was thinking of adding a --version flag that behaves like how `cargo install --version` does

**Comment by alsuren** [comment]
It seems that you beat me to creating a PR by 2 minutes (#61). I think you have some good ideas that I'm missing though.

File: cargo-quickinstall/src/main.rs (hunk 1 above)
Comment (alsuren) [reply cargo-quickinstall/src/main.rs:293]: I wonder if we could make a ["--print-version", "-V"] flag that does this.

Comment (progga) [reply cargo-quickinstall/src/main.rs:293]: I suspected that :)

**Comment by progga** [comment]
No problem David.  Feel free to ignore this one :)

File: cargo-quickinstall/src/main.rs [hunk 2]
```diff
@@ -285,14 +286,21 @@ fn install_crate(crate_name: &str, version: &str, target: &str) -> Result<(), In
 }
 
 fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
-    let args = std::env::args().collect::<Vec<_>>();
-    let crate_name = if let Some(true) = args.get(1).map(|a| a == "quickinstall") {
-        args.get(2)
```
Comment (alsuren) [inline cargo-quickinstall/src/main.rs:290]: I should have put a comment here. This hack is to handle how cargo calls out to subcommands. I did a vaguely equivalent thing in my attempt at this, over at:

https://github.com/alsuren/cargo-quickinstall/pull/61/files#diff-30cdea9081430e1fcfa8eea480f74e13b0cb145aaea84f5809a778db1c5cc08cR29

(I'm not sure what the best way to do this is. I might have to ask the maintainer of pico-args about it)

**Comment by alsuren** [comment]
> No problem David. Feel free to ignore this one :)

I feel like this was mostly my fault for not showing up to the event on time.

I think I will rebase my PR on top of yours, and incorporate the "remaining args" idea and a --print-version flag. Could you take a look at my PR and see if it makes sense?

I've also earmarked tomorrow for hacking on cargo-quickinstall. Is there anything that you would like to work on?

**Comment by alsuren** [comment]
I think it would be valuable to have a --print-version flag (-V for short)

**Comment by progga** [comment]
> I think it would be valuable to have a --print-version flag (-V for short)

Okay, I will give this a try.

**Comment by alsuren** [comment]
I think it would be valuable to have a --print-version flag (-V for short), so I will keep this PR open.

**Comment by progga** [comment]
Replaced by #65

**Pull request closed without merging**